zebra-consensus/src/block/subsidies.rs

```diff
@@ -0,0 +1,241 @@
+//! Calculation of Block Subsidy, Funding Streams, and Founders’ Reward
+
+use super::*;
+
+use zebra_chain::{
+    block::Height,
+    parameters::{Network, NetworkUpgrade::*},
+};
+
+use crate::parameters::{fs, Params};
+
+fn slow_start_shift() -> Height {
+    Height(Params::SLOW_START_INTERVAL / 2)
+}
+
+fn slow_start_rate() -> u32 {
+    Params::MAX_BLOCK_SUBSIDY / Params::SLOW_START_INTERVAL
+}
+
+fn is_bolossom_activated(height: block::Height, network: Network) -> Option<Height> {
+    let blossom_height = Blossom.activation_height(network)?;
+    let condition = height >= blossom_height;
+    match condition {
+        true => Some(blossom_height),
+        false => None,
+    }
+}
+
+fn halving(height: Height, network: Network) -> u32 {
+    let blossom_height = is_bolossom_activated(height, network);
+    let condition = height < slow_start_shift();
+    match condition {
+        true => 0,
+        false => match blossom_height {
+            Some(blossom_height) => {
+                let scaled_halvings = ((blossom_height.0 - slow_start_shift().0)
+                    * Params::BLOSSOM_POW_TARGET_SPACING_RATIO)
+                    + (height.0 - blossom_height.0);
+                scaled_halvings / Params::POST_BLOSSOM_HALVING_INTERVAL
+            }
+            _ => {
+                (((height.0 - slow_start_shift().0) / Params::PRE_BLOSSOM_HALVING_INTERVAL) as f32)
+                    .floor() as u32
```

Comment:
as far as I can tell this `floor` will have no effect, given that you convert it from an integer to a float right before you round and convert back.

Reply:
This is already gone at https://github.com/ZcashFoundation/zebra/pull/1051/commits/1530c46193958a134a220f16a6b0ec7c4691c1bb

The remaining `floor()` calls in other functions are actually needed.